ckan/lib/mailer.py

```diff
@@ -141,7 +141,8 @@ def get_reset_link(user):
                    h.url_for(controller='user',
                              action='perform_reset',
                              id=user.id,
-                             key=user.reset_key))
+                             key=user.reset_key,
+                             qualified=True))
```

Comment:
@mattfullerton The `urljoin` function is joining `site_url` (which is the original bug reported, as it should be `ckan.site_url` ) with the result of `h.url_for`. If `url_for` is passed the `qualified=True` parameter, it will be constructed with the `site_url` already appended, so there is no need to join both. So we need to drop the urljoin and just return the output of `h.url_for`